### PR TITLE
Integrate ACRE Radio Spectator functionality

### PR DESCRIPTION
**Pull request by Sniperhid**
**What this Pull Request does**
- Address - #388 
- Adds Radio button to spectator menu for show/hiding radio menu
image
- By default radio selection interface is hidden for now while we experiment with this feature. Pressing the button will reveal it
- Keeps ACRE specific configs in the `acre` addon, Adds CBA events `tmf_spectator_displayOnLoad`, `tmf_spectator_toggleUI`, `tmf_spectator_radioButtonPressed` to allow other addons/mods to extend spectator.

### CODE REVIEW

File: addons/acre2/config.cpp [hunk 1]
```diff
@@ -22,3 +22,24 @@ class cfgPatches
...
+class acre_sys_spectator_RscRadios;
+class acre_sys_spectator_RscSpeaking;
+class RscControlsGroupNoScrollbars;
+
+// Extend spectator
+class EGVAR(spectator,dialog) {
+    class controls {
+        class TargetUnitInfo: RscControlsGroupNoScrollbars {
+            idc = SPECTATOR_IDC_RADIO_INFO;
+            h = "((7.1) * ((((safezoneW / safezoneH) min 1.2)/ 1.2)/ 25))";
```
Comment (Freddo3000) [inline addons/acre2/config.cpp:35]: `h = 7.1 * GUI_GRID_H;`

File: addons/acre2/config.cpp [hunk 2]
```diff
@@ -22,3 +22,24 @@ class cfgPatches
...
+            w = "16 * (((safezoneW / safezoneH) min 1.2) / 40)";
```
Comment (Freddo3000) [inline addons/acre2/config.cpp:36]: `w = 16 * GUI_GRID_W;`

File: addons/acre2/config.cpp [hunk 3]
```diff
@@ -22,3 +22,24 @@ class cfgPatches
...
+            x = "12.1 * (((safezoneW / safezoneH) min 1.2) / 40) + (safezoneX + (safezoneW - ((safezoneW / safezoneH) min 1.2))/2)";
```
Comment (Freddo3000) [inline addons/acre2/config.cpp:37]: `x = 12.1 * GUI_GRID_W + GUI_GRID_CENTER_X;`

File: addons/acre2/config.cpp [hunk 4]
```diff
@@ -22,3 +22,24 @@ class cfgPatches
...
+            y = "((21) * ((((safezoneW / safezoneH) min 1.2)/ 1.2)/ 25)+ (safezoneY + (safezoneH - (((safezoneW / safezoneH) min 1.2)/ 1.2))/2))";
```
Comment (Freddo3000) [inline addons/acre2/config.cpp:38]: `y = 21 * GUI_GRID_H + GUI_GRID_CENTER_Y;`

File: addons/spectator/dialog.hpp [hunk 5]
```diff
@@ -187,6 +188,16 @@ class GVAR(dialog)
       text = "\A3\ui_f\data\gui\Rsc\RscDisplayArsenal\voice_ca.paa";
       tooltip = "MUTE SPECTATORS";
     };
+    class TMF_SPECTATOR_RADIO: RscSpectatorShortcutButton {
+        idc = IDC_SPECTATOR_TMF_SPECTATOR_RADIO;
+        onButtonDown = "['radio',_this] call tmf_spectator_fnc_menuhandler;";
```
Comment (Freddo3000) [inline addons/spectator/dialog.hpp]: `onButtonDown = QUOTE([ARR_2('radio',_this)] call FUNC(menuhandler));`

**Review by headswe — APPROVED** [review]

File: addons/acre2/config.cpp (hunk 1 above)
Comment (Sniperhid) [reply addons/acre2/config.cpp:35]: This is the initial integration which will give us an opportunity to try this out, I think once we know how useful it is the final placement will need some work which can be the scope of a later PR.

Comment (Freddo3000) [reply addons/acre2/config.cpp:35]: No doubt, just that is easier to work with macros instead of the full safezone statement. GUI_GRID is identical.

**Pull request merged**